src/components/pages/Verify/index.tsx

```diff
@@ -0,0 +1,319 @@
+import { BackHandler } from "react-native";
+import * as React from "react";
+import { NavigationScreenProps } from "react-navigation";
+import { MapStateToProps, connect } from "react-redux";
+import { RNFirebase } from "react-native-firebase";
+import DropdownAlert from "react-native-dropdownalert";
+
+import { MemberId } from "@raha/api-shared/dist/models/identifiers";
+
+import { Container } from "../../shared/elements";
+import { VerifyCamera } from "./VerifyCamera";
+import { VideoPreview } from "../Camera/VideoPreview";
+import {
+  getLoggedInMember,
+  getGenericPrivateVideoRef
+} from "../../../store/selectors/authentication";
+import { Member } from "../../../store/reducers/members";
+import { RahaState } from "../../../store";
+import { generateToken } from "../../../helpers/token";
+import { SubmitVerification } from "./SubmitVerification";
+import { VerifySplash } from "./VerifySplash";
+import { getMemberById } from "../../../store/selectors/members";
+import { getRequestVerificationOperation } from "../../../store/selectors/operations";
+import { ConfirmExistingVerificationVideo } from "./ConfirmExistingVerificationVideo";
+
+enum VerifyStep {
+  SPLASH,
+  CONFIRM_EXISTING_VIDEO,
+  CAMERA,
+  VIDEO_PREVIEW,
+  CONFIRM_VERIFICATION
+}
+
+type ReduxStateProps = {
+  loggedInMember?: Member;
+  toMember?: Member;
+  inviteVideoToken?: string;
+};
+
+type OwnProps = NavigationScreenProps<{ toMemberId: MemberId }>;
+
+type VerifyProps = OwnProps & ReduxStateProps;
+
+type VerifyState = {
```

Comment:
If you wanna be real fancy and not need the existence checks for `videoUri`/statically type your way into correctness, you can define `VerifyState` more granularly:

```
type VerifyState = {
  step: VerifyStep.SPLASH | VerifyStep.CAMERA
} | {
  step: VerifyStep.CONFIRM_EXISTING_VIDEO,
  inviteVideoToken: string
} | {
  step: VerifyStep.VIDEO_PREVIEW,
  videoUri: string
} // and so on
```

But that's not necessary per se and may be a little complex when it interacts with `props` as in `inviteVideoToken`, just double checks your work.